api/storage_service.cc

```diff
@@ -73,8 +73,12 @@ sstring validate_keyspace(const http_context& ctx, sstring ks_name) {
     throw bad_param_exception(replica::no_such_keyspace(ks_name).what());
 }
 
-sstring validate_keyspace(const http_context& ctx, const parameters& param) {
-    return validate_keyspace(ctx, param["keyspace"]);
+sstring validate_keyspace(const http_context& ctx, const std::unique_ptr<http::request>& req) {
+    return validate_keyspace(ctx, req->get_path_param("keyspace"));
+}
+
+sstring validate_keyspace(const http_context& ctx, const http::request& req) {
```

Comment:
why not expose only this function and let the other callers dereference their `std::unique_ptr<http::request>` to call this function?

Reply:
yeah, i concur with Benny on this.

Reply:
If I remember correctly, it saved me from having to modify another few dozens callers :-(

Reply:
i see. will come up with a follow up after this fix lands.